chain/client/src/client_actor.rs

```diff
@@ -266,16 +266,16 @@ impl Handler<WithSpanContext<NetworkClientMessages>> for ClientActor {
         msg: WithSpanContext<NetworkClientMessages>,
         ctx: &mut Context<Self>,
     ) -> Self::Result {
-        let msg_type: &str = (&msg.msg).into();
+        let WithSpanContext { msg, context, .. } = msg;
+        let msg_type: &str = (&msg).into();
         let span = tracing::debug_span!(
             target: "client",
             "handle",
             handler = "NetworkClientMessages",
             actor = "ClientActor",
             msg_type)
         .entered();
-        span.set_parent(msg.context);
-        let msg = msg.msg;
+        span.set_parent(context);
```

Comment:
Any reason this cannot use the `client_handler_span` helper?

Reply:
This invocation specifies `msg_type`. Extended the macro to handle it as well.